server/state/index.js

```diff
@@ -23,21 +23,30 @@ const initSmartContracts = () => {
   smartContracts = List()
 }
 
+const isDuplicateSmartContract = smartContract => {
+  if (smartContracts.find(x => x.smartContract === smartContract)) {
+    return true
+  } else return false
+}
+
 const addSmartContract = async smartContractObj => {
   const invalidFields = findInvalidSmartContractFields(smartContractObj)
-  if (invalidFields.length > 0)
+  if (invalidFields.length > 0) {
     throw new Error(
       `the following fields are missing or invalid: ${invalidFields.join(', ')}`
     )
+  }
+  if (isDuplicateSmartContract(smartContractObj.smartContract)) {
```

Comment:
Do you think we should be throwing an error or returning some specific response to the client? We don't want the server to error out because we'll lose the global state... 

Personally I think we should send a `409` back to the client https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409

just returning a 201? I was thinking if we're already listening we just return a 201 but i'm open to other strategies.